physics/SFC_Models/Land/Noahmp/noahmpdrv.F90

```diff
@@ -100,9 +119,309 @@ subroutine noahmpdrv_init(lsm, lsm_noahmp, me, isot, ivegsrc, &
+  integer                                   , intent(in) :: km         !vertical soil layer dimension 
+  integer,                                    intent(in) :: ncols
+  integer, intent(in)                                    :: isot
+  integer, intent(in)                                    :: ivegsrc
+
+  integer             , dimension(:)     , intent(in)    :: soiltyp    ! soil type (integer index)
+  integer             , dimension(:)     , intent(in)    :: vegtype    ! vegetation type (integer index)
+  real(kind=kind_phys), dimension(:)     , intent(inout) :: weasd      ! water equivalent accumulated snow depth [mm]
+  
+  type(land_iau_control_type)            , intent(inout) :: Land_IAU_Control
+  type(land_iau_external_data_type)      , intent(inout) :: Land_IAU_Data
+  type(land_iau_state_type)              , intent(inout) :: Land_IAU_State
+  real(kind=kind_phys), dimension(:,:)   , intent(inout) :: stc        ! soiltemp [K] 
+  real(kind=kind_phys), dimension(:,:)   , intent(inout) :: slc        !liquid soil moisture [m3/m3]'
+  real(kind=kind_phys), dimension(:,:)   , intent(inout) :: smc        !
+  character(len=*),                          intent(out) :: errmsg
+  integer,                                   intent(out) :: errflg
+  real(kind=kind_phys), intent(in)                       :: con_g       ! grav
+  real(kind=kind_phys), intent(in)                       :: con_t0c     ! tfreez
+  real(kind=kind_phys), intent(in)                       :: con_hfus    ! hfus
+
+  ! IAU update  
+  real(kind=kind_phys),allocatable, dimension(:,:)       :: stc_inc_flat, slc_inc_flat
+  real(kind=kind_phys), dimension(km)                    :: dz ! layer thickness
+  
+!TODO: This is hard-coded in noahmpdrv
+  real(kind=kind_phys)          :: zsoil(4) = (/ -0.1, -0.4, -1.0, -2.0 /)   !zsoil(km)
+
+  integer                       :: lsoil_incr
+  integer, allocatable          :: mask_tile(:)
+  integer,allocatable           :: stc_updated(:), slc_updated(:)
+  logical                       :: soil_freeze, soil_ice
+  integer                       :: soiltype, n_stc, n_slc
+  real(kind=kind_phys)          :: slc_new
+
+  integer                  :: i, j, ij, l, k, ib
+  integer                  :: lensfc
+
+  real(kind=kind_phys)                     :: smp !< for computing supercooled water 
+  real(kind=kind_phys)                     :: hc_incr
+
+  integer                  :: nother, nsnowupd
+  integer                  :: nstcupd, nslcupd, nfrozen, nfrozen_upd
+
+  !  --- Initialize CCPP error handling variables
+  errmsg = ''
+  errflg = 0
+ 
+  if (.not. Land_IAU_Control%do_land_iau) return
+
+  !> update current forecast hour  
+  ! GFS_control%jdat(:) = jdat(:)  
+  Land_IAU_Control%fhour=fhour
+  if(Land_IAU_Control%me == Land_IAU_Control%mpi_root) then 
+    print*,"itime ",itime," GFScont%fhour ",fhour," IauCon%fhour",Land_IAU_Control%fhour,   &
```

Comment:
Please consider wrapping any `print` statements in a debug logical that one can set in code or elsewhere. There have been complaints from folks that physics schemes add too much to the output console, cluttering it up.